client/src/App.tsx

```diff
@@ -1,18 +1,56 @@
 import React from "react";
-import { Route, BrowserRouter as Router } from "react-router-dom";
-import Home from "./ehr_components/home";
-import Main from "./ehr_components/main";
+import {
+  Route,
+  BrowserRouter as Router,
+  Redirect,
+  Switch
+} from "react-router-dom";
 import "bootstrap/dist/css/bootstrap.min.css";
 import "./App.css";
 
+import Home from "./ehr_components/home";
+import Main from "./ehr_components/Main";
+import Loader from "./components/Loader";
+import Auth from "./components/Auth";
+
+import { AuthContextProvider } from "./store/auth-context";
+import { useAuth } from "./hooks/auth-hook";
+
 const App: React.FC = () => {
-  return (
-    <Router>
-      <div className="App">
+  const { userData, login, logout, checkedStorage } = useAuth();
+
+  if (!checkedStorage) {
+    return <Loader />;
+  }
+
+  let routes;
+  if (userData.token) {

```

Comment:
Another way to do this is wrap all routes with AuthContextProvider
and then have  <PrivateRoute /> and <Route />

` export default function PrivateRoute({ component: Component, ...rest }) {
  const { isLoggedIn} = useAuth();

  return (
    <Route
      {...rest}
      render={props => {
        return isLoggedIn ? (
          <Component {...props} />
        ) : (
          <Redirect to="/login" />
        );
      }}
    ></Route>
  );
}`